internal/config/testdata/nginx-agent.conf

```diff
@@ -15,4 +15,12 @@ data_plane_config:
 client: 
   timeout: 10s
 
+metrics:
```

Comment:
The structure of the metrics config could do with being refactored a bit I think.
I dont think the boolean field `collector` is needed.

Would a structure like this be cleaner:
```
metrics:
  collector:
    otlp_export_url: "localhost:3000"
    otlp_receiver_url: "localhost:4318"
    receivers:
      - hostmetrics
    config_path: "/var/etc/nginx-agent/nginx-agent-otelcol.yaml"
```

Also for the urls do we want to specify if they are `http` or `https`?

Reply:
Hmm yeah, I'm unsure whether the `collector` namespace even necessary, as we will only support metrics via the collector going forward (I believe). Regardless, will try to clean up the config structure a bit (also working off @oliveromahony's [comment](https://github.com/nginx/agent/pull/698#discussion_r1631251782)).

Reply:
As discussed yesterday, `Metrics` has now been renamed to `Collector` and we are parsing a set of `Exporter`s and `Receiver`s.